### PR TITLE
Fixed nil pointer issue during tx gas price estimation in keeper

### PR DESCRIPTION
**Pull request by begmaroman**

### CODE REVIEW

**Review by cmalec — APPROVED** [review]

File: core/services/keeper/upkeep_executer_test.go [hunk 1]
```diff
@@ -191,6 +191,58 @@ func Test_UpkeepExecuter_PerformsUpkeep_Happy(t *testing.T) {
...
 
+	// Copy of above test with GlobalEvmEIP1559DynamicFees set to true it will cause the upkeep executor to panic
+	t.Run("runs upkeep on triggering block number on EIP1559 chain GlobalEvmEIP1559DynamicFees true", func(t *testing.T) {
+		db, config, ethMock, executer, registry, upkeep, job, jpv2, txm, _, _, _ := setup(t)
+
+		config.Overrides.GlobalEvmEIP1559DynamicFees = null.BoolFrom(true)
+
+		gasLimit := upkeep.ExecuteGas + config.KeeperRegistryPerformGasOverhead()
+		//once you know where the root nil pointer dereference is youll need to fix these gas values
+		//so the mocks get what they expect based on GasPrice nil
```
Comment (infiloop2) [inline core/services/keeper/upkeep_executer_test.go:202]: Can you productionise the test please

File: core/services/keeper/upkeep_executer_test.go [hunk 2]
```diff
@@ -191,6 +191,58 @@ func Test_UpkeepExecuter_PerformsUpkeep_Happy(t *testing.T) {
 		txm.AssertExpectations(t)
 	})
 
+	// Copy of above test with GlobalEvmEIP1559DynamicFees set to true it will cause the upkeep executor to panic
+	t.Run("runs upkeep on triggering block number on EIP1559 chain GlobalEvmEIP1559DynamicFees true", func(t *testing.T) {
+		db, config, ethMock, executer, registry, upkeep, job, jpv2, txm, _, _, _ := setup(t)
+
+		config.Overrides.GlobalEvmEIP1559DynamicFees = null.BoolFrom(true)
```
Comment (infiloop2) [inline core/services/keeper/upkeep_executer_test.go:198]: Do we have KeeperCheckUpkeepGasPriceFeatureEnabled = true by default?

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Can you also please verify that this test was failing before your change and now passes

File: core/services/keeper/upkeep_executer_test.go (hunk 2 above)
Comment (begmaroman) [reply core/services/keeper/upkeep_executer_test.go:198]: It is true by default. Just to be explicit.

**Comment by begmaroman** [comment]
@infiloop2 verified, it panics. 

Actually, it should be tested by an already existing test but there is a bug - it does not enable EIP1559. Fixed an existing test and added a new one (both for EIP1559 and non-EIP1559).

File: core/services/keeper/upkeep_executer_test.go (hunk 1 above)
Comment (begmaroman) [reply core/services/keeper/upkeep_executer_test.go]: Tests are ready for review now.

File: core/services/keeper/upkeep_executer_test.go [hunk 3]
```diff
@@ -144,9 +148,60 @@ func Test_UpkeepExecuter_PerformsUpkeep_Happy(t *testing.T) {
 		txm.AssertExpectations(t)
 	})
 
+	t.Run("runs upkeep on triggering block number on non-EIP1559 chain", func(t *testing.T) {
+		db, config, ethMock, executer, registry, upkeep, job, jpv2, txm, _, _, _ := setup(t)
+
+		config.Overrides.GlobalEvmEIP1559DynamicFees = null.BoolFrom(false)
```
Comment (infiloop2) [inline core/services/keeper/upkeep_executer_test.go:154]: the two tests look the same to me with the change of this config. Could you parameterise the two tests to share the code. (e.g. https://github.com/smartcontractkit/chainlink/blob/develop/core/cmd/remote_client_test.go#L151-L162)

Comment (begmaroman) [reply core/services/keeper/upkeep_executer_test.go]: Well, it is kind of difficult to refactor all tests to the test-table approach. I've updated it to use the same function but with a different flag. Check again.

**Review by infiloop2 — APPROVED** [review]
looks good, thanks for fixing it properly

**Comment by pinebit** [comment]
Should not this target 1.3.0-rc0 branch?

**Review by pinebit — APPROVED** [review]

**Pull request merged**